pkg/networkpath/traceroute/tcp/utils_windows.go

```diff
@@ -7,133 +7,31 @@
 package tcp
 
 import (
-	"context"
+	"errors"
 	"fmt"
 	"net"
-	"sync"
-	"time"
 
 	"golang.org/x/net/ipv4"
 	"golang.org/x/sys/windows"
 
 	"github.com/DataDog/datadog-agent/pkg/networkpath/traceroute/common"
-	"github.com/DataDog/datadog-agent/pkg/util/log"
-	"github.com/google/gopacket/layers"
 )
 
-var (
-	recvFrom = windows.Recvfrom
-)
-
-// listenPackets takes in raw ICMP and TCP connections and listens for matching ICMP
-// and TCP responses based on the passed in trace information. If neither listener
-// receives a matching packet within the timeout, a blank response is returned.
-// Once a matching packet is received by a listener, it will cause the other listener
-// to be canceled, and data from the matching packet will be returned to the caller
-func (w *winrawsocket) listenPackets(timeout time.Duration, localIP net.IP, localPort uint16, remoteIP net.IP, remotePort uint16, seqNum uint32) (net.IP, uint16, layers.ICMPv4TypeCode, time.Time, error) {
-	var icmpErr error
-	var wg sync.WaitGroup
-	var icmpIP net.IP
-	//var tcpIP net.IP
-	//var icmpCode layers.ICMPv4TypeCode
-	//var tcpFinished time.Time
-	var icmpFinished time.Time
-	var port uint16
-	wg.Add(1)
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-	go func() {
-		defer wg.Done()
-		defer cancel()
-		icmpIP, _, _, icmpFinished, icmpErr = w.handlePackets(ctx, localIP, localPort, remoteIP, remotePort, seqNum)
-	}()
-	wg.Wait()
-
-	if icmpErr != nil {
-		_, icmpCanceled := icmpErr.(common.CanceledError)
-		if icmpCanceled {
-			log.Trace("timed out waiting for responses")
-			return net.IP{}, 0, 0, time.Time{}, nil
-		}
-		if icmpErr != nil {
-			log.Errorf("ICMP listener error: %s", icmpErr.Error())
-		}
-
-		return net.IP{}, 0, 0, time.Time{}, fmt.Errorf("icmp error: %w", icmpErr)
+func (tp *TCPParser) MatchTCP(header *ipv4.Header, packet []byte, localIP net.IP, localPort uint16, remoteIP net.IP, remotePort uint16, seqNum uint32) (net.IP, error) {
```

Comment:
This could be in its own file, e.g. `tcp_parser_windows.go`.

Reply:
Took another look at this and the only windows specific stuff in here was able to be refactored so later I can go back and make the Linux implementation and Windows implementation match more closely.